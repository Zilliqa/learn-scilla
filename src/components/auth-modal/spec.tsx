import React from 'react';
import * as renderer from 'react-test-renderer';
import AuthModal from '.';
import { shallow } from 'enzyme';

const login = jest.fn();

const toggleAuthModal = () => console.log('toggle');
const t = (s: string) => s;

describe('Auth Modal tests', () => {
  const baseComponent = (props) => (
    <AuthModal
      toggleAuthModal={toggleAuthModal}
      login={login}
      isAuthModalOpen={false}
      isAuthPending={false}
      isLoaded={props.isLoaded}
      t={t}
    />
  );

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent({ isLoaded: false })).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the component while loading', () => {
      const wrapper = shallow(baseComponent({ isLoaded: false }));
      const assertion = wrapper.find('[data-test-id="auth-modal"]').length;
      expect(assertion).toBe(1);
    });

    it('renders the component after loaded', () => {
      const wrapper = shallow(baseComponent({ isLoaded: true }));
      const assertion = wrapper.find('[data-test-id="auth-modal"]').length;
      expect(assertion).toBe(1);
    });
  });

  describe('component behavior', () => {
    it('google login', () => {
      const wrapper = shallow(baseComponent({ isLoaded: true }));
      wrapper.find('[data-test-id="google-login-button"]').simulate('click');
      expect(login.mock.calls[0][0]).toBe('google');
      expect(login.mock.calls.length).toBe(1);
    });

    it('github login', () => {
      const wrapper = shallow(baseComponent({ isLoaded: true }));
      wrapper.find('[data-test-id="github-login-button"]').simulate('click');
      expect(login.mock.calls[1][0]).toBe('github');
      expect(login.mock.calls.length).toBe(2);
    });
  });
});
