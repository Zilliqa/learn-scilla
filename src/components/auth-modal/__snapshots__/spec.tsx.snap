// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly after loaded 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    link.signIn
  </a>
</li>
`;

exports[`renders correctly while loading 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    link.signIn
  </a>
</li>
`;
